src/Fixer/ConfigurationDefinitionFixerInterface.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer;
+
+use PhpCsFixer\OptionsResolver;
+
+/**
+ * @deprecated Will be merged into FixerInterface in 3.0
+ */
+interface ConfigurationDefinitionFixerInterface extends ConfigurableFixerInterface
+{
+    /**
+     * Defines the available configuration options of the fixer.
+     *
+     * @return OptionsResolver $configurationDefinition
+     */
+    public function getConfigurationDefinition();
+
+    /**
+     * Returns whether the fixer has configuration options.
+     *
+     * @return bool
+     */
+    public function isConfigurable();
```

Comment:
Then the getter (`getConfigurationDefinition` or sth) could throw Exception (as it will do it even if `isConfigurable` will return `false`, as both methods are public). Is it not ?